ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh

```diff
@@ -0,0 +1,382 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+workdir=`mktemp -d`
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+bastion_ignition_file="${workdir}/${CLUSTER_NAME}-bastion.ign"
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
+
+function patch_ignition_file()
+{
+  local base_ignition=$1
+  local patch_ignition=$2
+  t=$(mktemp)
+  # jq deepmerge 
+  # https://stackoverflow.com/questions/53661930/jq-recursively-merge-objects-and-concatenate-arrays
+  jq -s 'def deepmerge(a;b):
+  reduce b[] as $item (a;
+    reduce ($item | keys_unsorted[]) as $key (.;
+      $item[$key] as $val | ($val | type) as $type | .[$key] = if ($type == "object") then
+        deepmerge({}; [if .[$key] == null then {} else .[$key] end, $val])
+      elif ($type == "array") then
+        (.[$key] + $val | unique)
+      else
+        $val
+      end)
+    );
+  deepmerge({}; .)' "${base_ignition}" "${patch_ignition}" > "${t}"
+  mv "${t}" "${base_ignition}"
+  rm -f "${t}"
+}
+
+# base ignition
+cat > "${bastion_ignition_file}" << EOF
+{
+  "ignition": {
+    "config": {},
+    "security": {
+      "tls": {}
+    },
+    "timeouts": {},
+    "version": "3.0.0"
+  },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "sshAuthorizedKeys": [
```

Comment:
I think maybe we can use the original logic. set `"sshAuthorizedKeys": []`, then update it per `ssh_pub_key`, this wold help get some benefits when `ssh_pub_key` have multiple-lines ssh keys there.
```
# update ssh keys
tmp_keys_json=`mktemp`
tmp_file=`mktemp`
echo '[]' > "$tmp_keys_json"

readarray -t contents < "${ssh_pub_keys_file}"
for ssh_key_content in "${contents[@]}"; do
  jq --arg k "$ssh_key_content" '. += [$k]' < "${tmp_keys_json}" > "${tmp_file}"
  mv "${tmp_file}" "${tmp_keys_json}"
done

jq --argjson k "`jq '.| unique' "${tmp_keys_json}"`" '.passwd.users[0].sshAuthorizedKeys = $k' < "${bastion_ignition_file}" > "${tmp_file}"
mv "${tmp_file}" "${bastion_ignition_file}"
```